pkg/morph/client/notary.go

```diff
@@ -411,26 +415,95 @@ func (c *Client) NotarySignAndInvokeTX(mainTx *transaction.Transaction) error {
 		return err
 	}
 
-	// mainTX is expected to be pre-validated: second witness must exist and be empty
-	mainTx.Scripts[1].VerificationScript = multiaddrAccount.GetVerificationScript()
-	mainTx.Scripts[1].InvocationScript = append(
-		[]byte{byte(opcode.PUSHDATA1), 64},
-		multiaddrAccount.SignHashable(c.rpcActor.GetNetwork(), mainTx)...,
-	)
+	// Here we need to add a committee signature (second witness) to the pre-validated
+	// main transaction without creating a new one and without parsing transaction's
+	// signers. However, Notary actor demands the proper set of signers for constructor,
+	// thus, fill it from the main transaction's signers list.
+	//
+	// TODO: @roman-khimov, this list of signers is used only for notary Actor constructor,
+	//  after that it's useless. Thus, I'd skip the signers parsing down below and create
+	//  some fake account instead to pass the Actor constructor constraint.
+	s := make([]actor.SignerAccount, 0, 3)
+	s = append(s, []actor.SignerAccount{
+		{
+			// Proxy contract that will pay for the execution.
+			Signer:  mainTx.Signers[0],
+			Account: notary.FakeContractAccount(mainTx.Signers[0].Account),
+		},
+		{
+			// Inner ring multisignature.
+			Signer:  mainTx.Signers[1],
+			Account: multiaddrAccount,
+		},
+	}...)
+
+	if len(mainTx.Signers) > 3 {
+		// Invoker signature (simple signature account of storage node is expected).
```

Comment:
Then a fake one works fine as well, we're only supposed to be adding IR signatures now (we don't have any other cases for now), so we don't care who is the third signer, a correct signature is supposed to be present there anyway.

Reply:
The problem is that to create a proper `notary.FakeSimpleAccount(pub)` we need to parse existing verification script to get the `pub` out of it. The same problem with multisig account. And I'd like to completely avoid parsing, it's just not needed here and this signer is unesed anyway.